crates/wac-types/src/targets.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{ExternKind, ItemKind, SubtypeChecker, Types, WorldId};
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt;
+
+/// The result of validating a component against a target world.
+pub type TargetValidationResult = Result<(), TargetValidationReport>;
+
+/// The report of validating a component against a target world.
+#[derive(Debug, Default)]
+pub struct TargetValidationReport {
+    /// Imports present in the component but not in the target world.
+    pub imports_not_in_target: BTreeSet<String>,
+    /// Exports present in the target world but not in the component.
```

Comment:
Nit: It'd be nice if this was framed in the same way the `imports_not_in_target` field is documented (i.e., first the state of the component then the state of the target world):

```suggestion
    /// Exports not in the component but in the target world.
```